### PR TITLE
fix(web): further fixes BuildVisualKeyboard.  Fixes #2818

### LINKED ISSUE #2818: [web] BuildVisualKeyboard not working correctly after #2807
I've just tested this with 13.0.103 using https://github.com/keymanapp/help.keyman.com/pull/149 on http://help.keyman.com.local/keyboard/newa_traditional/1.0/newa_traditional and sadly it falls over:

```
Uncaught TypeError: Cannot set property 'className' of null
    at c.setButtonClass (visualKeyboard.ts:1402)
    at c._UpdateVKShiftStyle (visualKeyboard.ts:1371)
    at c.show (visualKeyboard.ts:1794)
    at Function.c.buildDocumentationKeyboard (visualKeyboard.ts:2188)
    at k.BuildVisualKeyboard (kmwbase.ts:599)
    at addKeyboard (kbd-docs.js:136)
    at addKeyboards (kbd-docs.js:177)
    at Object.success (kbd-docs.js:182)
    at j (jquery1-11-1.min.js:2)
    at Object.fireWith [as resolveWith] (jquery1-11-1.min.js:2)
```

_Originally posted by @mcdurdin in https://github.com/keymanapp/keyman/pull/2807#issuecomment-598461315_

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #2818.  This time, I made extra sure to test directly against the actual site's code - see https://github.com/keymanapp/help.keyman.com/pull/156.

### CODE REVIEW

File: web/source/osk/visualKeyboard.ts [hunk 1]
```diff
@@ -1871,7 +1915,7 @@ namespace com.keyman.osk {
       var b: HTMLElement = _Box, bs=b.style;
       bs.height=bs.maxHeight=oskHeight+'px';
 
-      b = <HTMLElement> b.childNodes.item(1).firstChild;
+      b = this.kbdDiv;
```
Comment (jahorton) [inline web/source/osk/visualKeyboard.ts]: This is literally the exact same object the original line was trying to reference.  That said, sometimes the old version accidentally accessed the predictive banner instead... so this is actually an improvement / fix for other aspects, too.

File: web/source/osk/visualKeyboard.ts [hunk 2]
```diff
@@ -1371,9 +1404,20 @@ namespace com.keyman.osk {
...
+        if(btn == null) {
+          //This can happen when using BuildDocumentationKeyboard, as the OSK isn't yet in the
+          //document hierarchy.  Sometimes.  (It's weird.)
+          btn = this.findKeyElement(layerId, states[i]);
+        }
+
+        if(btn != null) {
+          this.setButtonClass(keys[i], btn, this.layout);
+        } else {
+          console.warn("Could not find key to apply style: \"" + keyId + "\"");
```
Comment (darcywong00) [inline web/source/osk/visualKeyboard.ts]: If `btn` is still null at this point, isn't the keyID `states[i]` (l.1413)?

Comment (jahorton) [reply web/source/osk/visualKeyboard.ts]: KMW key ids are of the form `<layer>-<key-name>`.  Note the definition of `keyId` earlier in the code, which uses `states[i]` in place of `<key-name>`.

**Comment by darcywong00** [comment]
I checked the test pages
web\testing\build-visual-keyboard\index.html
web\bulk_rendering\index.html 

This is probably a separate issue, but still related to viewing OSK:

On the bulk_rendering page, it seemed all the layers looked identical (I only screenshot one keyboard)

bulk renderer

**Comment by jahorton** [comment]
Yep, separate issue, but easily fixed.  Kinda forgot to update a critical internal-API usage with v12.

File: web/source/osk/visualKeyboard.ts (hunk 1 above)
Comment (jahorton) [reply web/source/osk/visualKeyboard.ts]: After finding that this PR caused #2834, it turns out that this line is the culprit.  It's not `this.kbdDiv`, but instead `this.kbdDiv.firstChild` that the original line would have referred to.

**Pull request merged**